ksqldb-engine/src/test/java/io/confluent/ksql/physical/pull/operators/ProjectOperatorTest.java

```diff
@@ -0,0 +1,356 @@
+import io.confluent.ksql.logging.processing.ProcessingLogger;
+import io.confluent.ksql.metastore.MetaStore;
+import io.confluent.ksql.model.WindowType;
+import io.confluent.ksql.name.ColumnName;
+import io.confluent.ksql.planner.plan.ProjectNode;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.types.SqlTypes;
+import io.confluent.ksql.util.KsqlConfig;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.kstream.internals.TimeWindow;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ProjectOperatorTest {
+
+  private static final LogicalSchema SCHEMA = LogicalSchema.builder()
+      .keyColumn(ColumnName.of("k0"), SqlTypes.STRING)
+      .valueColumn(ColumnName.of("v0"), SqlTypes.STRING)
+      .valueColumn(ColumnName.of("v1"), SqlTypes.STRING)
+      .build();
+
+  private static final Struct A_KEY = StructKeyUtil
+      .keyBuilder(ColumnName.of("k0"), SqlTypes.STRING).build("k", 0);
+  private static final long A_ROWTIME = 12335L;
+
+  private static final Window A_WINDOW = Window.of(Instant.now(), Instant.now().plusMillis(10));
+  private static final TimeWindow STREAM_WINDOW = new TimeWindow(
+      A_WINDOW.start().toEpochMilli(),
+      A_WINDOW.end().toEpochMilli()
+  );
+
+  @Mock
+  private KsqlConfig ksqlConfig;
+  @Mock
+  private MetaStore metaStore;
+  @Mock
+  private ProcessingLogger logger;
+  @Mock
+  private Materialization mat;
+  @Mock
+  private LogicalSchema outputSchema;
+  @Mock
+  private SelectValueMapperFactorySupplier selectValueMapperFactorySupplier;
+  @Mock
+  private ProjectNode logicalNode;
+  @Mock
+  private AbstractPhysicalOperator child;
+  @Mock
+  private KsqlTransformer<Object, GenericRow> transformer;
+  @Mock
+  private SelectValueMapper<Object> selectValueMapper;
+
+  @Test
+  public void shouldProjectAllColumnsWhenSelectStarNonWindowed() {
+    // Given:
+    final ProjectOperator projectOperator = new ProjectOperator(
+        ksqlConfig,
+        metaStore,
+        logger,
+        mat,
+        logicalNode,
+        outputSchema,
+        true,
+        false,
+        false,
+        selectValueMapperFactorySupplier);
+    projectOperator.addChild(child);
+    final Row row = Row.of(
+        SCHEMA,
+        A_KEY,
+        GenericRow.genericRow("a", "b"),
+        A_ROWTIME
+    );
+    when(child.next()).thenReturn(row);
+    projectOperator.open();
+
+    // Then:
+    final List<Object> rowList = new ArrayList<>();
+    row.key().schema().fields().stream().map(row.key()::get).forEach(rowList::add);
+    rowList.addAll(row.value().values());
+    assertThat(projectOperator.next(), is(rowList));
```

Comment:
I think it makes sense to add an assert that `selectValueMapper` was never called (`Mockito.verifyNoInteractions(selectValueMapper)`). This will (1) make sure we're not calling unnecessary code, and (2) makes the test easier to understand (I asked myself why we didn't need to mock out the mapper for this method)